### PR TITLE
Parametrize ~~`newEval`~~ (now `processDeploys`) to enable charging selectively

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
Needed for https://rchain.atlassian.net/browse/RCHAIN-3281

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 1]
```diff
@@ -329,11 +329,7 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
     val initHashBlake = blakeFromByteString(initHash)
     terms.toList
       .foldM[F, Acc]((initHashBlake, Seq.empty))(evalSingle)
-      .map {
-        case (hash, deployResults) =>
-          val hashByteString = blakeToByteString(hash)
-          (hashByteString, deployResults)
-      }
+      .map(_.bimap(blakeToByteString, x => x))
```
Comment (JosDenmark) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: Just as an observation, `x => x` could also be `identity`.

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 2]
```diff
@@ -300,25 +300,23 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
         )
     )
 
-  private def newEval(
+  private def processDeploys(
       terms: Seq[DeployData],
-      runtime: Runtime[F],
-      initHash: StateHash
+      initHash: StateHash,
+      processDeploy: (Blake2b256Hash, DeployData) => F[(Blake2b256Hash, InternalProcessedDeploy)]
```
Comment (JosDenmark) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: 👏 👏 👏

**Comment by ArturGajowy** [comment]
bors try

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 3]
```diff
@@ -413,6 +412,9 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
     )
   }
 
+  private def blakeFromByteString[R](hash: StateHash): Blake2b256Hash =
```
Comment (lukasz-golebiewski) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: These two helpers belong in `Blake2b256Hash`

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 4]
```diff
@@ -356,10 +356,13 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
               //TODO: compare replay deploy cost to given deploy cost
               EvaluateResult(cost, errors) = replayEvaluateResult
               cont <- DeployStatus.fromErrors(errors) match {
-                       case int: InternalErrors => Left((Some(deploy), int)).pure[F]
+                       case int: InternalErrors =>
```
Comment (lukasz-golebiewski) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: What was the compilation error? Was sbt able to successfully compile it?

Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: There were many. You can checkout prior to this commit and see for yourself (in IDEA). Obviously yes.

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala (hunk 3 above)
Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: Done.

**Pull request merged**